state/execution.go

```diff
@@ -145,7 +145,9 @@ func (blockExec *BlockExecutor) Commit(block *types.Block) ([]byte, error) {
 		"appHash", fmt.Sprintf("%X", res.Data))
 
 	// Update mempool.
-	if err := blockExec.mempool.Update(block.Height, block.Txs); err != nil {
+	maxBytes := state.ConsensusParams.TxSize.MaxBytes
+	filter := func(tx types.Tx) bool { return len(tx) <= maxBytes }
```

Comment:
Maybe better to define this and export it.

Reply:
Also do we want to allow a negative or zero MaxBytes to indicate unbounded?

Reply:
It seems we're not checking TxSize.MaxBytes for txs in the block, just in the mempool. Technically that would make it not a consensus param, because a proposer running different mempool code could include txs larger than TxSize.MaxBytes

I wonder if we even need a TxSize.MaxBytes in the consensus params -  it might be simpler to just use the computed maxDataBytes. 

We'd still need `Update` to take a filter, but we'd be able to remove the TxSize configuration and just do it all through blocks.

Reply:
> though I can't find a super convincing reason as to why.

Right. We have a lot of surface area. When we don't have a really good reason to have something, it's probably worth removing it. Helps simplify things over all and reduce the possibility for errors.

Reply:
> Maybe better to define this and export it.

That was an initial implementation, but the problem is in https://github.com/tendermint/tendermint/blob/fef6ab6b79dc5d8fdfa1307cee8cb394fc00f05d/state/services.go#L26 it's undesirable to import mempool package (just for `mempl.Filter`)

Reply:
> We'd still need Update to take a filter, but we'd be able to remove the TxSize configuration and just do it all through blocks.

Should this be done along with enforcing MaxGas?